modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/EmptyResultMessagePropsTransformer.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {openSimpleInputModal} from 'frontend-js-web';
+
+function addAssetListEntry(itemData, namespace) {
```

Comment:
We can simplify this by placing `addAssetListEntry` function inside `propsTrasformer`. After the move, it doesn't need to have arguments.

Reply:
Yes we can that's true, and I'll do it.
That said, for me that's not simplifying because by doing that, we'll have a function inside a function instead of having separate functions - but that's just my opinion, I know React components and "propTransformers" are full of "inner" functions.

Reply:
It is a good practice to keep functions and variables in the lowest possible scope. It's a moot point in a simple file like this one, but things can get quickly out of hand the higher scope you use, topping with global :scream:

Reply:
> It is a good practice to keep functions and variables in the lowest possible scope.

Yes that's true.

I think I might have not explained myself correctly:

As far as I'm concerned, this 

```js
function foo() {


    function bar() {}
    function baz() {}
}
```

Is worse than this 

```js
function bar() {}

function baz() {}

function foo() {}
```

And I'm saying that because: 
- In the first example `bar` and `baz` are only visible inside `foo` and can't be re-used 
    (I know it doesn't matter in this case but it could)
- Also each time you're going to call `foo` in the first example you're re-creating the `bar` and `baz` functions ... and that has a cost, and since it's not necessary why should we do it?

> It's a moot point in a simple file like this one, but things can get quickly out of hand the higher scope you use, topping with global scream

That's not going to happen in this case, because this is a JavaScript module.

My point being: 

- It's easier to read a top level function than a nested function (inside a function)
- Re-creating functions on each function call as a higher cost than declaring them once 
- There's not risk of polluting the global scope because we're in a JavaScript module.

Reply:
All that to say that I didn't change this

Reply:
On a second though, there's not really any need for another function at the moment. So I updated that [here](https://github.com/liferay-frontend/liferay-portal/blob/cadea195c9899bdc873e7350a4ec7c4899785018/modules/apps/asset/asset-list-web/src/main/resources/META-INF/resources/js/EmptyResultMessagePropsTransformer.js#L31-L37).